PennMobile/HomeViewController.swift

```diff
@@ -0,0 +1,312 @@
+        
+        let event5 = Event(name: "GSWS101", location: "WILL 027", startTime: Time(hour: 1, minutes: 0, isAm: false), endTime: Time(hour: 2, minutes: 0, isAm: false))
+        
+        let event6 = Event(name: "CIS160", location: "MOOR 100", startTime: Time(hour: 7, minutes: 0, isAm: true), endTime: Time(hour: 2, minutes: 0, isAm: false))
+        
+        let event7 = Event(name: "PennQuest", location: "Houston Hall", startTime: Time(hour: 8, minutes: 30, isAm: true), endTime: Time(hour: 10, minutes: 30, isAm: true))
+        
+        return [event1, event2, event3, event4, event5, event6, event7]
+    }()
+    
+    //Weather for weather cell
+    var weather = Weather(temperature: "43", description: "sunny")
+    
+    //Study spaces for Reservation Cell, given in order that they appear in cell
+    var studySpaces: [StudyLocation] = {
+        let location1: StudyLocation = {
+            var location = StudyLocation(name: "Education Commons")
+            location.loadGSRs(for: [229, 221, 250])
+            return location
+        }()
+        
+        let location2: StudyLocation = {
+            var location = StudyLocation(name: "Van Pelt Library")
+            location.loadGSRs(for: [229, 221, 250])
+            return location
+        }()
+        
+        return [location1, location2]
+    }()
+    
+    //Announcement for Agenda Cell
+    var agendaAnnouncement: String? //= "Advanced Registration begins 3 days."
+    
+    //Spacing between cells
+    let cellSpacingHeight: CGFloat = 60
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.title = "Home"
+        self.screenName = "Home Page" //sets tracking for Google Analytics
+        
+        let image = UIImage(named: "homepage-settings")
+        navigationItem.rightBarButtonItem = UIBarButtonItem.itemWith(colorfulImage: image, color: UIColor(r: 100, g: 100, b:  100), target: self, action: #selector(handleShowSettings))
+        
+        //Table View
+        
+        registerCells()
+        
+        tableView.separatorStyle = .none //removes the separator lines between cells
+        tableView.showsVerticalScrollIndicator = false //removes the scroll bar
+        
+        //enables refresh on pulldown
+        refreshControl = UIRefreshControl()
+        refreshControl?.isEnabled = true
+        refreshControl?.addTarget(self, action: #selector(refreshAllData), for: .valueChanged)
+        refreshControl?.backgroundColor = .clear
+        refreshControl?.tintColor = .black
+        refreshControl?.attributedTitle = NSAttributedString(string: "Pull to refresh")
+        
+        tableView.addSubview(refreshControl!)
+        
+        refreshAllData(refreshControl!)
+        
+        //Default settings for dining halls
+        diningHalls = generateDiningHalls(for: ["1920 Commons", "English House", "Tortas Frontera", "New College House"])
```

Comment:
So much hardcoding everywhere, can these at least be consolidated into 1 location or even better, not hardcode these values?